### PR TITLE
Basic group support

### PR DESCRIPTION
**Pull request by liggitt**
- [x] Add groups field to user object
- [x] Add group name validation
- [x] Populate groups in UserFor mapping methods
- [x] Populate groups in OAuth token authenticator
- [x] Add tests for setting and clearing groups
- [x] Add tests for policy using groups specified on a user object

### CODE REVIEW

**Comment by liggitt** [comment]
[test]

**Comment by liggitt** [comment]
@deads2k fyi

File: pkg/user/api/types.go [hunk 1]
```diff
@@ -12,6 +12,8 @@ type User struct {
 	FullName string
 
 	Identities []string
+
+	Groups []string
```
Comment (deads2k) [inline pkg/user/api/types.go]: this should be a `StringSet`

Comment (liggitt) [inline pkg/user/api/types.go]: no, you may love writing custom conversion code, but I do not

**Comment by deads2k** [comment]
I think that covers the last item in your checklist.

lgtm.  merge if you agree.

**Comment by liggitt** [comment]
[merge]

**Pull request merged**